admin-site/src/Occasion.svelte

```diff
@@ -6,159 +6,113 @@
-export let broadcastStatus;
-
-let grabbedFromServerEvents;
-
-let formattedStartTimeFull;
-let formattedEndTimeFull;
-let formattedStartTime;
-let formattedEndTime;
-
-let isOccasionOpen;
-occasionOpen.subscribe(value => {
-  isOccasionOpen = value;
-})
-
-//double check that a delete Occasion has happened
-let deleteOccasionHasHappened = false;
-
-let cueState = 0;
-
-
-
-
-onMount(async () => {
-  formattedStartTimeFull = moment(focusedOccasion.startDateTime)
-    .format("LLL");
-  formattedEndTimeFull = moment(focusedOccasion.endDateTime)
-    .format("LLL");
-  formattedStartTime = moment(focusedOccasion.startDateTime)
-    .format("LL");
-  formattedEndTime = moment(focusedOccasion.endDateTime)
-    .format("LL");
-
-  urlStore.subscribe(value => {
-    serverURL = value;
-  })
-
-});
-
-
-  function openOccasionButton() {
+  
+  let focusedOccasion;
+  focusedOccasionStore.subscribe(value => {
+    focusedOccasion = value
+  });
+ 
+  let sliderCue = focusedEvent.episodes[0].cues[0];
+  export let broadcastStatus;
+
+
+  let formattedStartTimeFull;
+  let formattedEndTimeFull;
+  let formattedStartTime;
+  let formattedEndTime;
+
+  //double check that a delete Occasion has happened
+  let deleteOccasionHasHappened = false;
+
+  let cueState = 0;
+
+  onMount(async () => {
+    formattedStartTimeFull = moment(focusedOccasion.startDateTime)
+      .format("LLL");
+    formattedEndTimeFull = moment(focusedOccasion.endDateTime)
+      .format("LLL");
+    formattedStartTime = moment(focusedOccasion.startDateTime)
+      .format("LL");
+    formattedEndTime = moment(focusedOccasion.endDateTime)
+      .format("LL");
+  });
+
+  function serverCallToOpenOrCloseOccasion(state){
     try {
       fetch(serverURL + "/occasions/" + focusedOccasionID, {
         method: 'PATCH',
         headers: {'Content-Type': 'application/json'},
-        body: JSON.stringify({"state":"opened"}) 
+        body: JSON.stringify({"state": state}) 
       }).then( response => { 
         if(response.status == 200){
-          response.json().then( details => {
-            occasionOpen.set(true);
             // make sure store updates from server
             getEventsAndStore();
-          })
         } else {
           response.text().then( errorMessage => {
-            console.log('occasion open error on request: ' + errorMessage)
+            console.log('Error on ' + state + ' request: ' + errorMessage)
           })
         }
       }).catch( error => {
-        console.log("Error occasion open")
+        console.log("Catch error on " + state + ' request')
       })
     } catch (e) {
       console.log(e.message)
-    } 
-    //storedEvents.subscribe(value => console.log(value)) 
-    
+    }
+  }
+
+  function openOccasionButton() {
+    serverCallToOpenOrCloseOccasion("opened"); 
   };
 
- function closeOccasionButton(){
+  function closeOccasionButton(){
+    serverCallToOpenOrCloseOccasion("closed");  
+  };
+
+
+  function deleteOccasion() {
```

Comment:
I know we have at least one error case that relates here — if the occasion is open it can't be deleted. Does this get handled anywhere?

Reply:
While the occasion is open, currently there is no option to delete the occasion in the UI (only in its closed state). If the user tries to delete the event with an open occasion, that throws an error telling them which occasion is open.